modules/geochemistry/src/actions/AddTimeDependentReactionSolverAction.C

```diff
@@ -0,0 +1,218 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AddTimeDependentReactionSolverAction.h"
+#include "FEProblem.h"
+
+registerMooseAction("GeochemistryApp", AddTimeDependentReactionSolverAction, "setup_mesh");
+registerMooseAction("GeochemistryApp", AddTimeDependentReactionSolverAction, "init_mesh");
+registerMooseAction("GeochemistryApp", AddTimeDependentReactionSolverAction, "create_problem");
+registerMooseAction("GeochemistryApp", AddTimeDependentReactionSolverAction, "add_output");
+registerMooseAction("GeochemistryApp",
+                    AddTimeDependentReactionSolverAction,
+                    "add_geochemistry_molality_aux");
+registerMooseAction("GeochemistryApp",
+                    AddTimeDependentReactionSolverAction,
+                    "add_geochemistry_reactor");
+
+InputParameters
+AddTimeDependentReactionSolverAction::validParams()
```

Comment:
Same thing here about crazy code duplication for input params!

Reply:
Yep, done that, thanks